src/Ocelot/LoadBalancer/LoadBalancers/RoundRobin.cs

```diff
@@ -30,6 +30,17 @@ public async Task<Response<ServiceHostAndPort>> Lease(HttpContext httpContext)
                     }
 
                     var next = services[_last++];
+
+                    if (next == null)
+                    {
+                        return new ErrorResponse<ServiceHostAndPort>(new ServiceIsInvalidError($"The selected service was null in {nameof(RoundRobin)} during {nameof(Lease)} operation."));
+                    }
+
+                    if (next.HostAndPort == null)
+                    {
+                        return new ErrorResponse<ServiceHostAndPort>(new ServiceIsInvalidError($"The {nameof(next.HostAndPort)} was null in the selected service in {nameof(RoundRobin)} during {nameof(Lease)} operation."));
```

Comment:
As previously discussed, the `ServicesAreNullError` class can also be repurposed for this scenario.